plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/features/modelsourcewatcher/ModelSourceWatcher.java

```diff
@@ -0,0 +1,32 @@
+/********************************************************************************
+ * Copyright (c) 2020 EclipseSource and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+package org.eclipse.glsp.server.features.modelsourcewatcher;
+
+import org.eclipse.glsp.server.model.GModelState;
+
+public interface ModelSourceWatcher {
+
+   default void startWatching(final GModelState modelState) {}
```

Comment:
Please add some documentation above those methods. To me it is not obvious what each of those methods does and how they interact, e.g., what is the difference between starting and stopping or pausing and continuing? Is there any expected behavior when you call 'continueWatching' before it has been started, etc.

Reply:
Agreed, done!